biojava-alignment/src/main/java/org/biojava/nbio/alignment/SimpleGapPenalty.java

```diff
@@ -30,9 +32,13 @@
  *
  * @author Mark Chapman
  */
-public class SimpleGapPenalty implements GapPenalty {
+public class SimpleGapPenalty implements GapPenalty, Serializable {
```

Comment:
Hi Michael,

Thanks for reaching out. And you're right there are potential issues with
doing this.

I'd be interested to hear (offline or on here) how you've tackled this in
AMPLab.

Best wishes,

Anthony

On Tue, Mar 8, 2016 at 8:55 PM Michael L Heuer <notifications@github.com>
wrote:

> In
> biojava-alignment/src/main/java/org/biojava/nbio/alignment/SimpleGapPenalty.java
> <https://github.com/biojava/biojava/pull/427#discussion_r55472056>:
>
> > @@ -30,9 +32,13 @@
> >   *
> >   * @author Mark Chapman
> >   */
> > -public class SimpleGapPenalty implements GapPenalty {
> > +public class SimpleGapPenalty implements GapPenalty, Serializable {
>
> I saw the commit messages "Make classes serializable for Spark" earlier
> and reached out to see if might help (I currently work for the UC Berkeley
> AMPLab on genomics applications for Apache Spark).
>
> Making these classes serializable in this way is not only unsafe, and may
> cause future problems with API compatibility, it is not necessarily the
> most efficient way to use Biojava classes in Spark. A lot of static and
> non-threadsafe state including matrices and caches will also be serialized,
> leading to significant overhead and potential concurrency problems.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biojava/biojava/pull/427/files#r55472056>.
>